apps/api/src/expenses/expenses.controller.ts

```diff
@@ -46,11 +62,60 @@ export class ExpensesController {
   }
 
   @Delete(':id')
+  remove(@Param('id') id: string) {
+    return this.expensesService.remove(id)
+  }
+
+  @Get('campaign/:slug')
   @Roles({
     roles: [RealmViewSupporters.role, ViewSupporters.role],
     mode: RoleMatchingMode.ANY,
   })
-  remove(@Param('id') id: string) {
-    return this.expensesService.remove(id)
+  async listCampaignExpenses(@Param('slug') slug: string) {
+    return this.expensesService.listCampaignExpenses(slug)
+  }
+
+  @Get('campaign/approved/:slug')
+  @Public()
+  async listCampaignApprovedExpenses(@Param('slug') slug: string) {
+    return this.expensesService.listCampaignApprovedExpenses(slug)
+  }
```

Comment:
the route name could be more rest-like if named:
expenses/campaign/:slug/approved to indicate that the slug is of campaign, because currently it reads as if only approved campaigns will be returned.

Reply:
well the actual query from the front-end is like this:

/expenses/campaign/approved/:slug

The prefix is used to detect the controller.
I will leave it like this.